lesson07.md

```diff
@@ -0,0 +1,363 @@
+---
+
+After:
+
+```js
+variable++
+```
+
+Around:
+
+```js
+(value)
+```
+
+---
+
+Operators are also limited in the type of input values they take
+
+- Arithmetic: only numbers
+- Assignment: any type
+- Comparison: only between the same types, i.e. string to string
+- Logical: only booleans
+- String: only strings
+
+---
+
+When we perform an operation many times, we have to repeat the whole expression all the time
+
+```js
+person1Age * 10 + person1Height
+
+person2Age * 10 + person2Height
+
+person3Age * 10 + person3Height
+```
+
+---
+
+In producing new information from existing data,
+
+native operators are useful when
+
+- type of operation
+- type of input values
+- number of input values
+- code duplication
+
+are not an issue for us.
+
+---
+
+However if we want to bypass these limitations, we need to define our <span style="color: indianred;">own operators</span>.
+
+In Abdullah’s game, hypothetically we can think of an operator, which takes 2 input values, and produces the score
+
+```js
+person1Age @ person1Height
+```
+
+---
+
+### Increasing Code’s ability
+
+The custom operator we define can be used like this
+
+`score(person1Age, person1Height)`<br><br>
+
+`score`: is the name of our operator
+
+`person1Age`: is the first input value we pass
+
+`person1Height`: is the second input value we pass
+
+---
+
+When working with custom operators, the order of input values matters a lot
+
+```js
+score(person1Age, person1Height) !== 
+  score(person1Height, person1Age)
+```
+
+Contrast this to arithmetic operators
+
+```js
+2 + 3 === 3 + 2
+```
+
+---
+
+### Functions
+
+How do we define score operator?
+
+```js
+function score(age, height) {
+   return age * 10 + height;
+}
+```
+
+In JS, a custom operator is called a <span style="color: indianred;">function</span>
+
+---
+
+How do we define any operator?<br><br>
+
+```js
+function <operator>(<input1>, <input2>, …, <inputn>) {
+   // some logical computation
+   return <result>;
+}
+```
+
+---
+
+### Exercise
+
+Now it is your turn to implement score function.<br><br>
+
+First define score function which takes 2 input values<br><br>
+
+`   1st input value must be age`
+
+`   2nd input value must be height`<br><br>
+
+---
+
+Second use score function
+
+`  once with your own values`
+
+`  once with your classmate’s values`
+
+---
+
+Hint 1: to define a score function with two input values; age and height, follow
+
+```js
+function score(age, height) {
+ return age * 10 + height;
+}
+```
+
+---
+
+Hint 2: to use score function, follow
+
+```js
+score(42, 166);
+score(30, 175);
+```
+
+---
+
+### Exercise 2
+
+Now implement score2 function. 
+
+`  1st input value must be height`
+
+`  2nd input value must be age`
+
+Use it like this
+
+`  score2(166, 42);`
+
+---
+
+### Questions
+
+You have <span style="color: cadetblue;">implemented</span> two functions: score and score2.
+
+Tell the difference you observe, in the way you typed the <span style="color: indianred;">parameters list</span>
```

Comment:
maybe we should explicitly explain what's the "parameters list" before asking questions here. Even just a line in one of the previous slides where the input values are the "parameters list"

Reply:
I also believe that these terms need to be introduced formally. In the questions, all I wanted them to look at the highlighted code pieces and empirically tell that there are changes and no changes and be aware that function parts have names, which they need to study deeper at a later point.